lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1378,6 +1420,180 @@ func (w *WalletKit) fundPsbtInternalWallet(account string,
 	}, nil
 }
 
+// fundPsbtCoinSelect uses the "new" PSBT funding method using the channel
+// funding coin selection algorithm that allows specifying custom inputs while
+// selecting coins.
+func (w *WalletKit) fundPsbtCoinSelect(account string, changeIndex int32,
+	packet *psbt.Packet, minConfs int32, changeType ChangeAddressType,
+	feeRate chainfee.SatPerKWeight) (*FundPsbtResponse, error) {
+
+	// Before we select anything, we need to calculate the input, output and
+	// current weight amounts. While doing that we also ensure the PSBT has
+	// all the required information we require at this step.
+	var (
+		inputSum, outputSum btcutil.Amount
+		estimator           input.TxWeightEstimator
+	)
+	for i := range packet.Inputs {
+		in := packet.Inputs[i]
+
+		err := btcwallet.EstimateInputWeight(&in, &estimator)
+		if err != nil {
+			return nil, fmt.Errorf("error estimating input "+
+				"weight: %w", err)
+		}
+
+		inputSum += btcutil.Amount(in.WitnessUtxo.Value)
+	}
+	for i := range packet.UnsignedTx.TxOut {
+		out := packet.UnsignedTx.TxOut[i]
+
+		err := estimator.AddOutputWeight(out.PkScript)
+		if err != nil {
+			return nil, fmt.Errorf("error estimating output "+
+				"weight: %w", err)
+		}
+
+		outputSum += btcutil.Amount(out.Value)
+	}
+
+	// The amount we want to fund is the total output sum plus the current
+	// fee estimate. Since we pass the estimator of the current transaction
+	// into the coin selection algorithm, we don't need to subtract the fees
+	// here.
+	fundingAmount := outputSum - inputSum
+
+	var changeScriptLen int
+	switch {
+	case changeIndex >= 0:
+		changeOutput := packet.UnsignedTx.TxOut[changeIndex]
+		changeScriptLen = len(changeOutput.PkScript)
+
+	case changeType == ChangeAddressType_CHANGE_ADDRESS_TYPE_P2TR:
```

Comment:
Will P2TR be the default change type here?

Reply:
I would say no. Because the default for the normal `FundPsbt` is P2WKH (which was kept to not break backward compatibility with earlier RPC users, see more context in discussion here: https://github.com/lightningnetwork/lnd/pull/7348).
And since the change type is on the parent message, I don't think that using the new third funding option should introduce a different behavior of a top-level parameter.
But since we can specify the change type to be P2TR, I don't think the default is a big deal.